src/oneD/StFlow.cpp

```diff
@@ -344,190 +354,262 @@ void StFlow::updateProperties(size_t jg, double* x, size_t jmin, size_t jmax)
+    const double c_H2O[6] = {-0.23093, -1.12390, 9.41530, -2.99880,
+                                    0.51382, -1.86840e-5};
+    const double c_CO2[6] = {18.741, -121.310, 273.500, -194.050,
+                                    56.310, -5.8169};
+
+    // Calculation of the two boundary values
+    double boundary_Rad_left = m_epsilon_left * StefanBoltz * pow(T(x, 0), 4);
+    double boundary_Rad_right = m_epsilon_right * StefanBoltz * pow(T(x, m_points - 1), 4);
+
+    for (size_t j = jmin; j < jmax; j++) {
+        // calculation of the mean Planck absorption coefficient
+        double k_P = 0;
+        // Absorption coefficient for H2O
+        if (m_kRadiating[1] != npos) {
+            double k_P_H2O = 0;
+            for (size_t n = 0; n <= 5; n++) {
+                k_P_H2O += c_H2O[n] * pow(1000 / T(x, j), (double) n);
             }
-            // absorption coefficient for CO2
-            if (m_kRadiating[0] != npos) {
-                double k_P_CO2 = 0;
-                for (size_t n = 0; n <= 5; n++) {
-                    k_P_CO2 += c_CO2[n] * pow(1000 / T(x, j), (double) n);
+            k_P_H2O /= k_P_ref;
+            k_P += m_press * X(x, m_kRadiating[1], j) * k_P_H2O;
+        }
+        // Absorption coefficient for CO2
+        if (m_kRadiating[0] != npos) {
+            double k_P_CO2 = 0;
+            for (size_t n = 0; n <= 5; n++) {
+                k_P_CO2 += c_CO2[n] * pow(1000 / T(x, j), (double) n);
+            }
+            k_P_CO2 /= k_P_ref;
+            k_P += m_press * X(x, m_kRadiating[0], j) * k_P_CO2;
+        }
+
+        // Calculation of the radiative heat loss term
+        double radiative_heat_loss = 0;
+        radiative_heat_loss = 2 * k_P *(2 * StefanBoltz * pow(T(x, j), 4)
+                              - boundary_Rad_left - boundary_Rad_right);
+
+        // set the radiative heat loss vector
+        m_qdotRadiation[j] = radiative_heat_loss;
+    }
+}
+
+void StFlow::evalContinuity(double* x, double* rsd, int* diag,
+                            double rdt, size_t jmin, size_t jmax)
+{
+    // The left boundary has the same form for all cases.
+    if (jmin == 0) { // left boundary
+        rsd[index(c_offset_U,jmin)] = -(rho_u(x,jmin + 1) - rho_u(x,jmin))/m_dz[jmin]
+                                      -(density(jmin + 1)*V(x,jmin + 1)
+                                      + density(jmin)*V(x,jmin));
+    }
+
+    if (jmax == m_points - 1) { // right boundary (same for unstrained/free-flow)
+        rsd[index(c_offset_U, jmax)] = rho_u(x, jmax) - rho_u(x, jmax - 1);
+    }
+
+    // j0 and j1 are constrained to only interior points
+    size_t j0 = std::max<size_t>(jmin, 1);
+    size_t j1 = std::min(jmax, m_points - 2);
+    for (size_t j = j0; j <= j1; j++) {
+        rsd[index(c_offset_U, j)] = rho_u(x, j) - rho_u(x, j - 1);
+        diag[index(c_offset_U, j)] = 0;
+    }
+
+    if (m_usesLambda == false && m_isFree == false) { //unstrained flow
+        return;
+    }
```

Comment:
This is `unstrained-flow`. The way this is coded, it is evaluated for all cases (i.e. unnecessary for `free-flow` and `axisymmetric-flow`). I'd recommend moving it to the bottom as in the suggestion I posted yesterday.

Reply:
I feel like the name 'axisymmetric flow' would make more sense as strained flow or something else, because I believe all of these cases are using the axisymmetric governing equations from the Kee book?

Reply:
Technically true, but two cases turn out to be strictly 1D. Also, please refer to 

https://github.com/Cantera/cantera/blob/2f3ba6a577b406c60151480fcd12a27994992b7c/src/oneD/StFlow.cpp#L124-L132